lib/views/post/widgets/vote_button.dart

```diff
@@ -0,0 +1,170 @@
+    return _VoteButtonState();
+  }
+}
+
+class _VoteButtonState extends State<VoteButton> with TickerProviderStateMixin {
+  AnimationController rotationController;
+  AnimationController colorController;
+  AnimationController favoriteController;
+  Animation<Color> colorTween;
+  Animation<double> favoriteMorph;
+  CurvedAnimation favoriteMorphCurveAnimation;
+  Color color;
+  Color lastColor;
+  IconData icon;
+
+  onStateChange(VoteAnimation voteAnimation) {
+    setState(() {
+      lastColor = color;
+      color = _getColorByAnimation(voteAnimation);
+
+      if (voteAnimation == VoteAnimation.voteFocused ||
+          voteAnimation == VoteAnimation.voteUnfocused) {
+        if (widget.type == VoteButtonType.favorite) {
+          if (widget.animationService.initialVote == Vote.favorite)
+            favoriteController.reverse();
+          else
+            favoriteController.forward();
+        } else
+          rotationController.forward();
+      } else if (voteAnimation == VoteAnimation.clearFocused ||
+          voteAnimation == VoteAnimation.clearUnfocused) {
+        if (widget.type == VoteButtonType.favorite) {
+          if (widget.animationService.initialVote != Vote.favorite)
+            favoriteController.reverse();
+          else
+            favoriteController.forward();
+        } else
+          rotationController.reverse();
+      }
+
+      if (lastColor != color) {
+        colorTween =
+            ColorTween(begin: lastColor, end: color).animate(colorController);
+        colorController.forward(from: 0);
+      }
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    lastColor = color = _getInitialColor();
+    rotationController = new AnimationController(
+      vsync: this,
+      duration: voteAnimationDuration,
+    );
+    colorController = new AnimationController(
+      vsync: this,
+      duration: voteAnimationDuration,
+    );
+    favoriteController = new AnimationController(
+      vsync: this,
+      duration: voteAnimationDuration,
+    );
+    widget.animationService.addButtonStateListener(widget.type, onStateChange);
+    icon = widget.type == VoteButtonType.up
+        ? Icons.add_circle_outline
+        : Icons.remove_circle_outline;
+    colorTween =
+        ColorTween(begin: lastColor, end: color).animate(colorController);
+    if (widget.type == VoteButtonType.favorite) {
+      icon = null;
+      favoriteMorphCurveAnimation = CurvedAnimation(
+          parent: favoriteController,
+          curve: Curves.easeOutCubic,
+          reverseCurve: Curves.easeInCubic);
+      favoriteMorph = Tween(
+              begin: widget.animationService.initialVote == Vote.favorite
+                  ? 1.0
+                  : 0.0,
+              end: widget.animationService.initialVote != Vote.favorite
+                  ? 1.0
+                  : 0.0)
+          .animate(favoriteMorphCurveAnimation);
+    }
+  }
+
+  @override
+  void dispose() {
```

Comment:
Don't you have to dispose some of your Animation(Controller)s?

Reply:
finally fixed with 6a1d672
other dispose are fixed with the split into mixins